### PR TITLE
Handle zero-like imaginary part as zero in to_r

### PR DESCRIPTION
**Pull request by mrkn**
Use imag.to_r when imag is not 0 or 0r.

Fixes [Bug #5179]

### CODE REVIEW

File: complex.c [hunk 1]
```diff
@@ -1851,8 +1851,11 @@ nucomp_to_r(VALUE self)
     get_dat1(self);
 
     if (!k_exact_zero_p(dat->imag)) {
-        rb_raise(rb_eRangeError, "can't convert %"PRIsVALUE" into Rational",
-                 self);
+        VALUE imag = rb_check_convert_type_with_id(dat->imag, T_RATIONAL, "Rational", idTo_r);
```
Comment (KJTsanaktsidis) [inline complex.c:1854]: why convert `imag` to rational here just to pass it into `k_exact_zero_p()`? Wouldn't it be easier to just change the check `k_exact_zero_p(dat->imag)` to `f_zero_p(dat->imag)` instead?

Comment (mrkn) [reply complex.c:1854]: @KJTsanaktsidis This approach doesn’t impose the responsibility of deciding what exactly constitutes zero on the Complex class. Instead, it assigns the responsibility to each individual class.

Comment (mrkn) [reply complex.c:1854]: And this implementation is for proof of concept. Before merging this to the master branch, I’ll add the optimization for Float class later.

Comment (KJTsanaktsidis) [reply complex.c:1854]: Ah that makes sense - thanks for the explanation!

**Comment by mrkn** [comment]
Waiting for https://github.com/ruby/prism/issues/2468.

**Comment by kddnewton** [comment]
Sorry about the delay @mrkn, you should be able to rebase and it should pass. Please let me know if you run into any more trouble!

**Pull request merged**